library/Ivoz/Provider/Domain/Model/SurvivalDevice/SurvivalDevice.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Ivoz\Provider\Domain\Model\SurvivalDevice;
+
+/**
+ * SurvivalDevice
+ */
+class SurvivalDevice extends SurvivalDeviceAbstract implements SurvivalDeviceInterface
+{
+    use SurvivalDeviceTrait;
+
+    /**
+     * @codeCoverageIgnore
+     * @return array<string, mixed>
+     */
+    public function getChangeSet(): array
+    {
+        return parent::getChangeSet();
+    }
+
+    /**
+     * Get id
+     * @codeCoverageIgnore
+     */
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    protected function sanitizeValues(): void
+    {
+        $this->udpPort = $this->sanitizePort($this->udpPort);
+        $this->tcpPort = $this->sanitizePort($this->tcpPort);
+        $this->tlsPort = $this->sanitizePort($this->tlsPort);
+        $this->wssPort = $this->sanitizePort($this->wssPort);
+
+        $this->setProxy($this->proxy);
+        $this->setOutboundProxy($this->outboundProxy);
+    }
+
+    private function sanitizePort(int $port): int
+    {
+        if ($port < 1 || $port > 65535) {
+            throw new \DomainException("Invalid port number: $port. Must be between 1 and 65535.");
+        }
+
+        return $port;
+    }
+
+    public function setUdpPort(int $udpPort): static
+    {
+        return parent::setUdpPort($this->sanitizePort($udpPort));
+    }
+
+    public function setTcpPort(int $tcpPort): static
+    {
+        return parent::setTcpPort($this->sanitizePort($tcpPort));
+    }
+
+    public function setTlsPort(int $tlsPort): static
+    {
+        return parent::setTlsPort($this->sanitizePort($tlsPort));
+    }
+
+    public function setWssPort(int $wssPort): static
+    {
+        return parent::setWssPort($this->sanitizePort($wssPort));
+    }
+
+    protected function setOutboundProxy(?string $outboundProxy = null): static
+    {
+        if (!is_null($outboundProxy)) {
+            $outboundProxy = trim($outboundProxy);
+            if ($outboundProxy === '') {
+                throw new \DomainException('Outbound Proxy cannot be empty.');
+            }
+            if ($this->getProxy() && strpos($this->getProxy(), ':') !== false) {
+                throw new \DomainException('Proxy must not include a port when Outbound Proxy is set.', 70003);
+            }
+        } else {
+            $this->setOutboundProxy(null);
+        }
```

Comment:
```suggestion
        if ($this->getOutboundProxy()) {
            $outbound_proxy = explode(':', $this->getOutboundProxy());
            $ip = array_shift($outbound_proxy);
            $obPort = array_shift($outbound_proxy);
            if (!is_null($port)) {
                throw new \DomainException('When Outbound Proxy is used, SIP Proxy must not include a port.', 70003);
            }
        }    
```

Or am I missing something?

Anyway you have an infinite recursive loop if _$outboundProxy_ is _null_.